### PR TITLE
feat(minimessage): selector tag

### PR DESCRIPTION
**Pull request by NoahvdAa**
Adds a tag for `SelectorComponent`s (part of  #678):

Example usage:
```
<sel:@a/>
<selector:@s/>
```

### CODE REVIEW

**Comment by MiniDigger** [comment]
(docs PR please)

looking good

**Review by kezz — CHANGES_REQUESTED** [review]
Waiting for an equiv docs PR, but otherwise, this LGTM! Thank you :)

**Review by zml2008 — APPROVED** [review]
Docs PR now present

File: text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/standard/SelectorTag.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    StandardTags.names(SEL, SELECTOR),
+    SelectorTag::create,
+    SelectorTag::claim
+  );
+
+  private SelectorTag() {
+  }
+
+  static Tag create(final ArgumentQueue args, final Context ctx) throws ParsingException {
+    final String key = args.popOr("A selection key is required").value();
```
Comment (zml2008) [inline text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/standard/SelectorTag.java:57]: There's now a `separator` parameter that selector components take, with a Component value, can you add that as a second, optional argument here?

**Review by zml2008 — CHANGES_REQUESTED** [review]

**Comment by NoahvdAa** [comment]
Sorry for not making a docs PR, I completely forgot. The separator has been added.

File: text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/standard/SelectorTag.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    return Tag.inserting(Component.selector(key, separator));
+  }
+
+  static @Nullable Emitable claim(final Component input) {
+    if (!(input instanceof SelectorComponent)) return null;
+
+    final SelectorComponent st = (SelectorComponent) input;
+    return emit -> {
+      emit.tag(SEL);
+      emit.argument(st.pattern());
```
Comment (kezz) [inline text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/standard/SelectorTag.java:72]: Should we be serializing the selector here too?

Comment (zml2008) [reply text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/standard/SelectorTag.java:72]: probably yeahhhh

**Review by zml2008 — APPROVED** [review]

**Pull request merged**